File: app/src/main/java/com/sinthoras/randograf/cards/Card.java
package com.sinthoras.randograf.cards;

import androidx.annotation.StringRes;
import androidx.fragment.app.Fragment;

public abstract class Card extends Fragment {

    protected Card(int contentLayerID) {
        super(contentLayerID);
    }

    protected boolean drawnWithRuin = false;

    public void notifyPriorRuin() {
        drawnWithRuin = true;
    }

    public abstract int getDuration();

    @StringRes
    public abstract int getTitle();
}

File: app/src/main/java/com/sinthoras/randograf/cards/Card.kt
package com.sinthoras.randograf.cards

import androidx.annotation.StringRes
import java.io.Serializable

abstract class Card: Serializable {

    private var drawnWithRuin = false

    fun notifyPriorRuin() {
        drawnWithRuin = true
    }

    fun getDrawWithRuin() = drawnWithRuin

    abstract fun getDuration(): Int

    @StringRes
    abstract fun getTitle(): Int
}